### PR TITLE
Update old validate script to Python 3

### PR DESCRIPTION
**Pull request by busterbeam**
I haven't removed `jsonschema` but that can be added as a module in `python3`.  Sorry for the confusing backward logic of True and False being swapped.  your exit status was that way around so... yea your fault.  Just trying to make the code shorter simpler and neater

### CODE REVIEW

File: bin/ckan-validate.py [hunk 1]
```diff
@@ -1,58 +1,50 @@
...
+                try:
+                    validate(load(ckan_file), schema)
+                    yield not print(f"Success!")
+                except ValidationError as error:
+                    yield print(
+                        "Failed! See below for error description.\n", error)
+                except ValueError as error:
+                    yield print(
+                        "Failed! This error will be cryptic,\n",
+                        "but often a JSOqN or property error", error)
```
[the remark this answers is not in the document]
Comment (busterbeam) [reply bin/ckan-validate.py:36]: Thanks must of fell asleep on my keyboard

**Comment by HebaruSan** [comment]
This script is pretty old and no longer in use, but I guess if we're not going to delete it, we might as well update it in case someone finds a use for it.

**Comment by HebaruSan** [comment]
Probably a good idea to test it first, though. :smirk: 

```
$ bin/ckan-validate.py ~/Downloads/KSP/CKAN-meta/Astrogator/Astrogator-v0.10.2.ckan 
Traceback (most recent call last):
  File "/home/user/Downloads/KSP/CKAN/bin/ckan-validate.py", line 47, in <module>
    else: exit(main(argv[1:]))
  File "/home/user/Downloads/KSP/CKAN/bin/ckan-validate.py", line 43, in main
    return all(ckan_validates(ckan_paths, schema))
TypeError: ckan_validates() takes 1 positional argument but 2 were given
```

File: bin/ckan-validate.py [hunk 2]
```diff
@@ -1,58 +1,49 @@
...
+            return schema
+
+def ckan_validates(ckan_paths):
+    for ckan_path in ckan_paths:
+        if exists(ckan_path):
+            with open(ckan_path, 'r') as ckan_file:
+                print(f"Validating {ckan_path}..")
+                try:
+                    validate(load(ckan_file), schema)
+                    yield not print(f"Success!")
```
Comment (HebaruSan) [inline bin/ckan-validate.py:29]: My linter is telling me `print` doesn't return a value, so it probably doesn't make sense to put it in an expression like this.

File: bin/ckan-validate.py [hunk 3]
```diff
@@ -1,58 +1,49 @@
...
-
-    sys.exit(error)
+    with open(schema_path, 'r') as schema_file:
+        schema = load(schema_file)
+        if schema == None:
+            return print("Could not parse JSON schema, exiting..")
+        else:
+            return schema
+
+def ckan_validates(ckan_paths):
```
Comment (HebaruSan) [inline bin/ckan-validate.py:22]: Looks like you meant to have `schema` as a second parameter here.

File: bin/ckan-validate.py (hunk 2 above)
Comment (HebaruSan) [inline bin/ckan-validate.py:29]: My linter is telling me `print` doesn't return a value, so it probably doesn't make sense to put it in an expression like this.

**Comment by HebaruSan** [comment]
> the confusing backward logic of True and False being swapped. your exit status was that way around

There's a module called `exitstatus` that's pretty good for this, you can return `ExitStatus.failure` or `ExitStatus.success` instead of hard coding numbers. Willing to try that?

**Comment by HebaruSan** [comment]
Looks like this isn't . Better to

**Comment by HebaruSan** [comment]
Looks like this isn't going to get updated. For now it's better to have an old, working script than an updated, broken one.

@busterbeam, if you have a chance to address the review comments here, please let us know and we can re-open this.

**Comment by HebaruSan** [comment]
No updates, closing again.

**Pull request closed without merging**